samtranslator/model/eventsources/push.py

```diff
@@ -727,7 +727,7 @@ def to_cloudformation(self, **kwargs):  # type: ignore[no-untyped-def]
         resources = []
 
         function = kwargs.get("function")
-        intrinsics_resolver = kwargs.get("intrinsics_resolver")
+        intrinsics_resolver: IntrinsicsResolver = kwargs["intrinsics_resolver"]
```

Comment:
Why haven't we been sure that it's not `None` but now we are?

Reply:
As I see below, we probably have to create an empty `IntrinsicResolver` here if we need it but... still we need to be very cautious about that. Is it `self.add_auth_to_swagger` which requires the resolver not to be `None`? Can we handle `None` value inside it?

Reply:
`self.add_auth_to_swagger` expects `IntrinsicsResolver` but the `.get()` is returning an `Optional[Any]`. 

I don't think we can handle this inside of `add_auth_to_swagger()` but it seems like if i add a check on `intrinsics_resolver` before calling `add_auth_to_swagger()` it is happy.

i.e. `if swagger_body and self.Auth and intrinsics_resolver and self.Auth.get("OverrideApiAuth"):`

Reply:
Sounds good. But do we really need an intrinsic resolver to be non null to make override happen?